contracts/test/Verifiers.ts

```diff
@@ -0,0 +1,184 @@
+import { expect } from "chai";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+import { ethers } from "hardhat";
+import { Verifier as SnarkVerifier, InclusionVerifier, GrandsumVerifier, Halo2VerifyingKey } from "../typechain-types";
+import { BigNumber } from "ethers";
+import { BytesLike } from "ethers/lib/utils";
+import * as fs from "fs";
+import * as path from "path";
+
+describe("Verifier Contracts", () => {
+  async function deployVerifyingFixture() {
+    // Contracts are deployed using the first signer/account by default
+    const verifyingKey = await ethers.deployContract(
+      "src/VerifyingKey.sol:Halo2VerifyingKey",
+    ) as Halo2VerifyingKey;
+
+    const commitmentJson = fs.readFileSync(path.resolve(__dirname, "../../kzg_prover/bin/commitment_solidity_calldata.json"), "utf-8");
+    const commitmentCalldata = JSON.parse(commitmentJson);
+
+    return {
+      verifyingKey,
+      commitmentCalldata,
+    };
+  }
+
+
+  describe("Snark Proof Verifier", () => {
+    let snarkVerifier: SnarkVerifier;
+    let verifyingKey: Halo2VerifyingKey;
+    let commitmentCalldata: {
+      range_check_snark_proof: BytesLike;
+      grand_sums_batch_proof: BytesLike;
+      total_balances: BigNumber[];
+    };
+    beforeEach(async () => {
+      const deploymentInfo = await loadFixture(deployVerifyingFixture);
+      verifyingKey = deploymentInfo.verifyingKey;
+      commitmentCalldata = deploymentInfo.commitmentCalldata;
+
+      // Deploy SnarkVerifier contract
+      snarkVerifier = await ethers.deployContract(
+        "src/SnarkVerifier.sol:Verifier"
+      ) as SnarkVerifier;
+
+      await snarkVerifier.deployed();
+    });
+
+    it("should verify snark proof", async () => {
+      // The verifier contract checks the number of instances in the VerifyingKey contract at 0x00c0 with the given 'instances' input
+      expect(await snarkVerifier.verifyProof(verifyingKey.address, commitmentCalldata.range_check_snark_proof, [1])).to.be.true;
+    });
+
+    it("should fail to verify snark proof without the number of instances", async () => {
```

Comment:
I think this test is not necessary because the instance issue should be fixed by merging the upstream changes.

Reply:
Okay, I will remove it.

Reply:
Replaced another test case instead of this at https://github.com/summa-dev/summa-solvency/commit/6db5fb7cc56192741c86d55ff8f59c9eb7dea05b